src/utils/objectUtils.spec.ts

```diff
@@ -371,4 +372,178 @@ describe('objectUtils', () => {
       )
     })
   })
+
+  describe('convertDateFieldsToIsoString', () => {
+    it('Empty object', () => {
+      expect(convertDateFieldsToIsoString({})).toStrictEqual({})
+    })
+
+    type TestInputType = {
+      id: number
+      value: string
+      date: Date
+      code: number
+      reason?: string | null
+      other?: TestInputType
+      array?: {
+        id: number
+        createdAt: Date
+      }[]
+    }
+
+    type TestExpectedType = {
+      id: number
+      value: string
+      date: string
+      code: number
+      other?: TestExpectedType
+      array?: {
+        id: number
+        createdAt: string
+      }[]
+    }
+
+    it('simple object', () => {
+      const date = new Date()
+      const input: TestInputType = {
+        id: 1,
+        date,
+        value: 'test',
+        reason: 'reason',
+        code: 100,
+      }
+
+      const output: TestExpectedType = convertDateFieldsToIsoString(input)
```

Comment:
why do we need to cast type here?

Reply:
The function is automatically inferring the type.

yes, it is like an assert to make sure the type is being inferred correctly